File: app/src/main/java/com/example/viper/presentation/modules/details/router/DetailsRouter.java
package com.example.viper.presentation.modules.details.router;

import android.app.Activity;

import javax.inject.Inject;

import com.example.viper.presentation.modules.details.contract.DetailsRouterContract;
import com.vershininds.mixture.router.AbstractRouter;
import com.vershininds.mixture.view.AndroidComponent;


public class DetailsRouter extends AbstractRouter<DetailsRouterContract.Presenter>
        implements DetailsRouterContract.Router {

    @Inject
    public DetailsRouter() {
    }

    @Override
    public void finishScreen(AndroidComponent androidComponent) {
        if (androidComponent != null) {
            Activity activity = androidComponent.getActivity();
            activity.finish();
        }
    }
}

File: app/src/main/java/com/vershininds/mixture/sample/presentation/modules/details/router/DetailsRouter.kt
package com.vershininds.mixture.sample.presentation.modules.details.router

import com.vershininds.mixture.dispatcher.ActionDispatcher
import com.vershininds.mixture.action.RouterAction
import com.vershininds.mixture.action.handle
import com.vershininds.mixture.dispatcher.subscribeRouter
import com.vershininds.mixture.router.MxtRouter2
import com.vershininds.mixture.sample.presentation.modules.details.contract.DetailsRouterContract
import com.vershininds.mixture.view.AndroidComponent
import javax.inject.Inject

class DetailsRouter @Inject constructor(
        private val androidComponent: AndroidComponent,
        private val dispatcher: ActionDispatcher
) : MxtRouter2 {

    private val tag = DetailsRouter::class.java.simpleName

    override fun subscribeOnDispatcher() {
        dispatcher.subscribeRouter(tag) { action: RouterAction ->
            when (action) {
                is DetailsRouterContract.TypeRouterAction.FinishScreenAction -> action.handle { finishScreen(androidComponent) }
            }
        }
    }

    override fun unsubscribeDispatcher() {
        dispatcher.removeSubscriber(tag)
    }

    private fun finishScreen(androidComponent: AndroidComponent) {
        androidComponent.activity.finish()
    }
}